dbt_schema_builder/relation.py

```diff
@@ -117,11 +118,16 @@ def is_unmanaged(self):
         """
         Is it "unmanaged" (i.e. it has been added to the list of unmanaged tables in
         unmanaged_tables.yml, indicating that we do not want schema builder to manage this table's
-        view-generating models), and
+        view-generating models)
         """
-        return (
-            "{}.{}".format(self.app, self.relation) in self.unmanaged_tables
-        )
+        for unmanaged_table in self.unmanaged_tables:
+            # make sure to include the EOL character in the regex, to prevent
+            # matching a substring in a larger string.
+            unmanaged_table_regex = re.compile(r'{}$'.format(unmanaged_table))
+            relation_name = "{}.{}".format(self.app, self.relation)
+            if re.search(unmanaged_table_regex, relation_name):
```

Comment:
I'm confused by this.  I understood that if you compiled a re (e.g. unmanaged_table_regex), that Pattern object would then have methods on it, like `unmanaged_table_regex.full_match(relation_name)`.  I thought that re.search took a pattern *string* as an object, i.e. the string that was input to re.compile.  Like https://docs.python.org/3/library/re.html#re.Pattern.match.  

Also, re.search will look for matches anywhere in the string, while match will look starting at the beginning.   Youj don't have a ^ here -- is that because the schema is included in the relation?  It's still possible that a table name has a different "prefix".

Reply:
Hmm, yes, I suppose it is possible. I can put in a fix & test for this.